src/JsonSerializationWriter.php

```diff
@@ -310,43 +310,62 @@ public function writeAnyValue(?string $key, $value): void{
         $type = get_debug_type($value);
         switch ($type) {
             case 'float':
-                $this->writeFloatValue($key, $value);
+                if (is_float($value))
```

Comment:
Nit..
If we need `is_float(), is_string() etc` to be 100% sure of the type, maybe we don't need to `get_debug_type` then switch? Perhaps only use if-else with is_float(), etc?